pkg/controller/generic_reconciler.go

```diff
@@ -94,6 +105,8 @@ func (gr *GenericReconciler) AddToManager(mgr manager.Manager) error {
 
 // Start validating the given object kind every interval.
 func (gr *GenericReconciler) Start(ctx context.Context) error {
+	go gr.ConfigChanged(ctx)
```

Comment:
Just an idea to consider here. What about moving the listener part directly to the validation engine and then update registry directly there?

Reply:
Well you would need to add the `Start` method to the validation engine and then register it in the manager as well.

Reply:
This change was implemented.